src/lib/classes/notification.js

```diff
@@ -0,0 +1,512 @@
+             */
+            const typedValue = item.type === 'NamedNode' ? namedNode(value) : literal(value);
+
+            writer.addQuad(
+              namedNode(''),
+              namedNode(`${context[item.property.split(':')[0]]}${item.label}`),
+              typedValue
+            );
+          }
+        } else {
+          throw new SolidError('Schema do not have property', 'Notification', 500);
+        }
+      });
+
+      await writer.end(async (error, result) => {
+        if (error) {
+          throw error;
+        }
+        /**
+         * Custom header options to create a notification file on pod.
+         * options:
+         * @slug: {String} custom file name that will be save it on the pod
+         * @contentType: {String} format of the file that will be save it on the pod.
+         */
+        const optionsHeader = options && options.header;
+
+        await solid.fetch(to, {
+          method: 'POST',
+          body: result,
+          headers: {
+            'Content-Type': 'text/turtle',
+            ...optionsHeader
+          }
+        });
+      });
+      return solidResponse(200, 'Notification was created');
+    } catch (error) {
+      throw new SolidError(error.message, 'Notification', error.status);
+    }
+  };
+
+  /**
+   * Mark as read the notifications
+   * @param notificationPath
+   * @returns {Promise<*>}
+   */
+  markAsRead = async (notificationPath, status) => {
+    try {
+      /**
+       * Update subject read into notification a notification file.
+       */
+      await solidLDflex[notificationPath]['https://www.w3.org/ns/solid/terms#read'].set(status);
+
+      return solidResponse(200, 'Notification was updated');
+    } catch (error) {
+      throw new SolidError(error.message, 'Notification', error.status);
+    }
+  };
+
+  /**
+   * Delete notification file from inbox folder and container list
+   * @param filename
+   * @param inboxRoot
+   * @returns {Promise<*>}
+   */
+  delete = async file => {
+    try {
+      /**
+       * Delete a file at an inbox folder by default will delete the reference on inbox container
+       */
+      await solid.fetch(file, { method: 'DELETE' });
+
+      return solidResponse(200, 'Notification was deleted it');
+    } catch (error) {
+      throw new SolidError(error.message, 'Notification Delete', error.status);
+    }
+  };
+
+  /**
+   * Get the predicate value from the shape object
+   * @param field
+   * @param shapeName
+   * @returns {string}
+   */
+  getPredicate = (field, shapeName) => {
+    const prefix = field.property.split(':')[0];
+    const ontology = this.schema[shapeName]['@context'][prefix];
+    return `${ontology}${field.label}`;
+  };
+
+  /**
+   * Find user's inbox from document file
+   * @param document file url to find inbox path
+   * @returns {Promise<boolean>}
+   */
+  discoveryInbox = async document => {
+    try {
+      const existDocument = await this.hasInbox(document);
```

Comment:
rename this to something like hasDocument or documentExists

Reply:
Done